### ISSUE TITLE
Attempting to update object and save in cli creates high CPU and memory before hanging

### ISSUE THREAD

**Issue by pspierce** [pimcore/pimcore#6789]
![2020-07-15_11-36](https://user-images.githubusercontent.com/456876/87565446-a63f2300-c68f-11ea-8b9d-b181ed7cd4b8.png)
# Bug Report

### Expected behavior
Using the cli, save a dataobject without the cli hanging.  Saving an object through the admin is fine.

### Actual behavior
Starting around 6.6.6 or so, I've had issues using the cli to save dataobjects.  The behavior has continued on 6.6.9. I followed the process through to Concrete.php and when $fd is a block, $fd->checkValidity goes into Block then BlockElement then into /var/www/pim/vendor/myclabs/deep-copy/src/DeepCopy/DeepCopy.php's recursiveCopy and once there it doesn't seem to come back.

### Steps to reproduce
Save the object as defined in the attached via the console.

### Code example
```php
<?php

namespace ProductBundle\Command;


use Pimcore\Console\AbstractCommand;
use Pimcore\Console\Dumper;
use Pimcore\Model\DataObject;
use Symfony\Component\Console\Input\InputInterface;
use Symfony\Component\Console\Output\OutputInterface;
use ProductBundle\EventListener\ProductListener;
use ZedBundle\Utils\ZedConnect;
use ZedBundle\ZedBundle;
use ProductBundle\Tools\Utils\DataObjectHierarchy;
use ProductBundle\Tools\Utils;

class ProductUpdateStatusCommand extends AbstractCommand
{

    protected function configure()
    {
        $this
            ->setName('product:updates:status')
            ->setDescription('Various pull product status from SAP.');
    }

    function _group_by($array, $key)
    {
        $return = array();
        foreach ($array as $val) {
            $return[$val[$key]][] = $val;
        }
        return $return;
    }

    protected function execute(InputInterface $input, OutputInterface $output)
    {
        if ($this->profiler !== null) {
            $this->profiler->disable();
        }

        \Pimcore\Db::getConnection()->getConfiguration()->setSQLLogger(null);
        $zed = new ZedConnect();
        $returnedRows = $zed->execQuery('Status', 'PIM', 'LIVE', true, true);
        $zed = null;
        $de = \Pimcore::getEventDispatcher();
        dump(array_keys($de->getListeners()));
        foreach ($de->getListeners() as $listener => $event) {
            $de->removeListener($listener, $event);
        };
        foreach ($returnedRows as $row) {
            $db = \Pimcore\Db::get();
            $id = $db->fetchCol("select oo_id from object_26 where o_type = 'variant' and ItemNumber = '" . $row['itemcode'] . "'");
            if (!$id[0]) {
                continue;
            }
            $obj = DataObject\Product::getById(
                $id[0],
                [
                    'limit' => 1,
                    'unpublished' => true,
                    'objectTypes' => [DataObject::OBJECT_TYPE_VARIANT]
                ]
            );
            echo $id[0] . ' .. ';
            if ($obj) {
                $save = false;
                if ($obj->getId() == 3442) {
                    continue;
                }
                if (!in_array($row['u_status'], ['Active', 'In Development']) && !$obj->getArchived()) {
                    $obj->setArchived(1);
                    $save = true;
                } elseif (in_array($row['u_status'], ['Active', 'In Development']) && $obj->getArchived()) {
                    $obj->setArchived(0);
                    $save = true;
                }
                if ($obj->getStatus() != $row['u_status']) {
                    $obj->setStatus($row['u_status']);
                    $save = true;
                }
                if ($save) {
                    ProductListener::removeListeners($obj);
                    print('here\n');
                    $obj->save();
                    print('returned\n');
                }
            }
            $obj = null;
        }
    }
}

```
Some of the above is awkward because I was trying to figure out how to get around this issue.  Oddly, there is no issue saving through the admin.

# Feature Request

**Comment by pspierce** [comment]
CSV Import appears to have similar issues.

**Comment by pspierce** [comment]
I believe I finally fixed it with:
\Pimcore\Model\DataObject\AbstractObject::setGetInheritedValues(false);
Something isn't right there, however.

**Comment by dpfaffenbauer** [comment]
I have quite a similar problem. The Pimcore Maintenance Task runs forever and consumes memory until the server dies, even though PHP is configured to only use 1GB of memory. I have honestly currently no idea what causes the problem.....

**Comment by brusch** [comment]
@pspierce can you maybe share the cachegrind file? Maybe that helps.

**Comment by pspierce** [comment]
The profile that I took the screenshot from is too large.  I'll try and recreate with something smaller.

**Comment by pspierce** [comment]
It goes into $data = $blockElement->getData(); in Block.php at line 1151 and then goes down a rabbit hole of recursiveCopy in DeepCopy.

**Comment by pspierce** [comment]
Here is the file I viewed in kcachegrind. https://www.dropbox.com/s/oh1kyzaoicbujki/crc32.tar.gz?dl=0

**Comment by pspierce** [comment]
Here is a trace: https://www.dropbox.com/s/yjokleu9th6229y/cachegrind.pricing.xt.tar.gz?dl=0

**Comment by weisswurstkanone** [comment]
Not sure what it is in the cachegrind.pricing file ? How to open it ?

**Comment by weisswurstkanone** [comment]
@pspierce can you provide a class definition export + some instructions on how to fill it with data ?

**Comment by weisswurstkanone** [comment]
@pspierce there was a change here https://github.com/pimcore/pimcore/pull/6236
but much earlier than 6.6.6, I believe in 6.6.2

Also I don't see how this could cause your troubles. Besides that I am not able to reproduce it with my data.
Maybe some interesting constellation in your data?

Would be nice if you could invest half and hour or so and try to find out why it does this recursion.
Otherwise we have to try and replicate your setup on my instance.

**Comment by pspierce** [comment]
It is a pretty involved object.  I've attached the definition for Product and the definition for the object in the block where it just stops.
[definitions.zip](https://github.com/pimcore/pimcore/files/4954145/definitions.zip)

**Comment by pspierce** [comment]
I've spent more than a half hour.  I've spent hours and hours.  I can't see it.  Pimcore is my first time writing in PHP so that doesn't help.  The strange thing that is a clue is the fact that the web interface doesn't appear to be impacted by this.  If I update one single value in a Product and $obj->save() then it hangs in the CLI.  I'm currently rewriting the updates I was using in the CLI to use the REST API due to urgency.

**Comment by pspierce** [comment]
Oh, and I did just recently upgrade from 6.6.2 and upgraded symfony from 4.3 to 4.4 as well.  I had held back symfony to 4.3 because of another bug.

**Comment by weisswurstkanone** [comment]
Please attach the JSON class export.

**Comment by pspierce** [comment]
If I run with extra verbose (-vvv) it just shows:
11:52:46 DEBUG     [cache] Successfully got data for key object_2545 from cache
[
  "key" => "object_2545"
]
11:52:46 DEBUG     [cache] Successfully got data for key object_2545 from cache
[
  "key" => "object_2545"
]
11:52:46 DEBUG     [cache] Successfully got data for key object_2545 from cache
[
  "key" => "object_2545"
]
11:52:46 DEBUG     [cache] Successfully got data for key object_2545 from cache
[
  "key" => "object_2545"
]
11:52:46 DEBUG     [cache] Successfully got data for key object_2545 from cache
[
  "key" => "object_2545"
]
11:52:46 DEBUG     [cache] Successfully got data for key object_14972 from cache
[
  "key" => "object_14972"
]
11:52:46 DEBUG     [cache] Successfully got data for key object_14951 from cache
[
  "key" => "object_14951"
]
11:52:46 DEBUG     [cache] Successfully got data for key object_2545 from cache
[
  "key" => "object_2545"
]
11:52:46 DEBUG     [cache] Successfully got data for key object_2545 from cache
[
  "key" => "object_2545"
]
11:52:46 DEBUG     [cache] Successfully got data for key asset_1705 from cache
[
  "key" => "asset_1705"
]
11:52:46 DEBUG     [cache] Successfully got data for key object_2545 from cache
[
  "key" => "object_2545"
]
11:52:46 DEBUG     [cache] Successfully got data for key object_2545 from cache
[
  "key" => "object_2545"
]
11:52:46 DEBUG     [cache] Successfully got data for key object_2545 from cache
[
  "key" => "object_2545"
]
11:52:46 DEBUG     [cache] Successfully got data for key object_2545 from cache
[
  "key" => "object_2545"
]
11:52:46 DEBUG     [cache] Successfully got data for key object_2545 from cache
[
  "key" => "object_2545"
]
11:52:46 DEBUG     [cache] Successfully got data for key object_2545 from cache
[
  "key" => "object_2545"
]
11:52:46 DEBUG     [cache] Successfully got data for key object_2545 from cache
[
  "key" => "object_2545"
]
11:52:46 DEBUG     [cache] Successfully got data for key object_2545 from cache
[
  "key" => "object_2545"
]
11:52:46 DEBUG     [cache] Successfully got data for key object_14972 from cache
[
  "key" => "object_14972"
]
11:52:46 DEBUG     [cache] Successfully got data for key object_14951 from cache
[
  "key" => "object_14951"
]
11:52:46 DEBUG     [cache] Successfully got data for key object_2545 from cache
[
  "key" => "object_2545"
]
11:52:46 DEBUG     [cache] Successfully got data for key object_2545 from cache
[
  "key" => "object_2545"
]
11:52:46 DEBUG     [cache] Successfully got data for key asset_1705 from cache
[
  "key" => "asset_1705"
]
11:52:46 DEBUG     [cache] Successfully got data for key object_2545 from cache
[
  "key" => "object_2545"
]
11:52:46 DEBUG     [cache] Successfully got data for key object_2545 from cache
[
  "key" => "object_2545"
]
11:52:46 DEBUG     [cache] Successfully got data for key object_2545 from cache
[
  "key" => "object_2545"
]
11:52:46 DEBUG     [cache] Successfully got data for key object_2545 from cache
[
  "key" => "object_2545"
]
11:52:46 DEBUG     [cache] Successfully got data for key object_2545 from cache
[
  "key" => "object_2545"
]
11:52:46 DEBUG     [cache] Successfully got data for key object_2545 from cache
[
  "key" => "object_2545"
]
11:52:46 DEBUG     [cache] Successfully got data for key object_2545 from cache
[
  "key" => "object_2545"
]
11:52:46 DEBUG     [cache] Successfully got data for key object_2545 from cache
[
  "key" => "object_2545"
]
11:52:46 DEBUG     [cache] Successfully got data for key object_14972 from cache
[
  "key" => "object_14972"
]
11:52:46 DEBUG     [cache] Successfully got data for key object_14951 from cache
[
  "key" => "object_14951"
]
11:52:46 DEBUG     [cache] Successfully got data for key object_2545 from cache
[
  "key" => "object_2545"
]
11:52:46 DEBUG     [cache] Successfully got data for key object_2545 from cache
[
  "key" => "object_2545"
]
11:52:46 DEBUG     [cache] Successfully got data for key asset_1705 from cache
[
  "key" => "asset_1705"
]
11:52:46 DEBUG     [cache] Successfully got data for key object_2545 from cache
[
  "key" => "object_2545"
]
11:52:46 DEBUG     [cache] Successfully got data for key object_2545 from cache
[
  "key" => "object_2545"
]
11:52:46 DEBUG     [cache] Successfully got data for key object_2545 from cache
[
  "key" => "object_2545"
]
11:52:46 DEBUG     [cache] Successfully got data for key object_2545 from cache
[
  "key" => "object_2545"
]
over and over.
[class.zip](https://github.com/pimcore/pimcore/files/4954639/class.zip)

**Comment by weisswurstkanone** [comment]
Log file looks ok.

However, I am afraid it is not that easy. Cannot import the product class because the workflow stuff (ProductBundle) is missing. Can you reduce the class to a minimum?

**Comment by weisswurstkanone** [comment]
Wait, I inspected the Product json file.
Do I see it correctly that there is a block inside a localized field inside a block? If so, this is not supported.

**Comment by pspierce** [comment]
I'll move that and see.  I don't think that is it, however.  One script actually worked pretty well this morning the first time or two and then started showing the same behavior.  Strange that using the admin via the web is not hanging like the CLI is.

**Comment by dpfaffenbauer** [comment]
> I have quite a similar problem. The Pimcore Maintenance Task runs forever and consumes memory until the server dies, even though PHP is configured to only use 1GB of memory. I have honestly currently no idea what causes the problem.....

Ok, I found my problem. Has nothing todo with Pimcore per-se, it was a combination of ImageMagick, Sqip and an odd svg.

**Comment by pspierce** [comment]
I deleted the block->localized->block in my dev system.  Same behavior.  Definitely something with blocks as that is where things hang.  If I change this:
![image](https://user-images.githubusercontent.com/456876/88321455-33682480-cced-11ea-81fb-1495de3f49eb.png)
to this:
![image](https://user-images.githubusercontent.com/456876/88321541-57c40100-cced-11ea-9541-e566c0b1d51f.png)

And all the problems go away.

**Comment by weisswurstkanone** [comment]
Sorry, still not able to reproduce this :-(

**Comment by pspierce** [comment]
I am still dealing with this and it appears to be growing and impacting other products whereas it was previously a small handful.  I updated my object class to get rid of the nested block->localizedfield->block and the issue continues.  I've attached the product.json and several stacktraces.  When the product hits BlockElement, it goes into DeepCopy and appears to be in an infinite loop.  If I go into mysql and delete the value in BlkProductBOMComponent then the product loads in the admin without issue.
[stacktrace2.txt](https://github.com/pimcore/pimcore/files/5489488/stacktrace2.txt)
[stacktrace1.txt](https://github.com/pimcore/pimcore/files/5489489/stacktrace1.txt)
[class_ComponentProductTypes_export.json.txt](https://github.com/pimcore/pimcore/files/5489494/class_ComponentProductTypes_export.json.txt)
[class_Product_export (1).json.txt](https://github.com/pimcore/pimcore/files/5489495/class_Product_export.1.json.txt)

**Comment by brusch** [comment]
Should be fixed by #7563 and #11561
